dropshot/src/lib.rs

```diff
@@ -568,7 +568,7 @@ pub(crate) struct ResponseInfo {
 }
 
 #[usdt::provider]
-mod dropshot {
+mod probes {
```

Comment:
Did this change the provider name?

Reply:
That would change the provider name, yes, to `probes`. If we want to specify the path at which the probe macros are available, the `probe_path` and `probe_name` arguments to the `usdt::provider` macro are the way to do that. See [here](https://docs.rs/usdt/0.1.19/usdt/#configurable-names) for a description of how to do that. cc @pfmooney

Reply:
To be explicit, if you want the probes available at a module called `crate::probes`, you'd do:

```rust
#[usdt::provider(probe_path = "probes")]
mod dropshot {
    ...
}
```

Reply:
Sorry, I misunderstood that documentation, since both dtrace probes and rust modules are `:`-delimited.  I know this would constitute another breaking change for usdt, but what about overriding the provider name via an argument to `#[usdt::provider]`, rather than one to override how the macros are exposed?  I know the module and the functions are essentially a fiction, consumed by the proc-macro to emit the real probe bits, but I think the advantage to your new module-scoped-probes implementation is that it more closely follows the ergonomics of normal rust structures:
You define some module (`probes`) and call the member functions (or in this case, macros) from it.  Having to name the module `dropshot` and then rename it from the proc-macro for how it exposes those probes to the source subverts those expectations, IMO.

Reply:
> To be explicit, if you want the probes available at a module called `crate::probes`, you'd do:
> 
> ```rust
> #[usdt::provider(probe_path = "probes")]
> mod dropshot {
>     ...
> }
> ```

Yeah, I understand it's possible.  I'm just suggesting that the naming be switched around in usdt so it be more like this:

```rust
#[usdt::provider(probe_provider = "dropshot")]
mod probes {
    ...
}
```

Reply:
@pfmooney To make sure I understand, you're suggesting that the name of the module is always what's written there, and that's how consumers refer to the generated probe macros. What can be changed, though, is the name of the provider -- it defaults to the name of the module, but you could override it. That right?

It seems reasonable to me, but I'd love @ahl's thoughts too. Definitely another breaking API change, so we'll have to hold this PR behind that, if it's desirable.

Reply:
Yeah, that the proc-macro not rename the module, but rather could be instructed to (re) name the provider as it is exposed to dtrace.

Also, I agree that this PR need to wait or be reworked to not rename the provider in the mean time.